projects/objectnav_baselines/experiments/robothor/objectnav_robothor_rgb_resnet18gru_ddppo.py

```diff
@@ -0,0 +1,62 @@
+from typing import Sequence, Union
+
+import torch.nn as nn
+
+from allenact.base_abstractions.preprocessor import Preprocessor
+from allenact.utils.experiment_utils import Builder, TrainingPipeline
+from allenact_plugins.ithor_plugin.ithor_sensors import (
+    GoalObjectTypeThorSensor,
+    RGBSensorThor,
+)
+from projects.objectnav_baselines.experiments.robothor.objectnav_robothor_base import (
+    ObjectNavRoboThorBaseConfig,
+)
+from projects.objectnav_baselines.mixins import (
+    ResNetPreprocessGRUActorCriticMixin,
+    ObjectNavPPOMixin,
+)
+
+
+class ObjectNavRoboThorRGBPPOExperimentConfig(ObjectNavRoboThorBaseConfig,):
```

Comment:
I guess the comma after the base config class is no longer needed.

Other than that, I'll use this comment to point out that, as much as I like having most of the config in a single file, I'm not sure what the problem with multiple inheritance was. As long as we follow a certain order, the latest overload should be the one in use, right? Just curious about what the issue was :)

Reply:
Very fair question :) the problem I was having was that the order of method resolution in multiple inheritance was causing me some issues when I tried to reuse some experiment configs across multiple projects. In particular, there was a `Base` config who returned `tuple()` for its set of preprocessors and I wanted to mix this with a `CLIPPreprocessorMixin` config that defined the clip preprocesor. The issue then was the if I put `Base` first, I wouldn't get the clip preprocessors (as it would return `tuple()`) but if I put `CLIPPreprocessorMixin` then it would go through that preprocessor's methods first and find that one of them had a `raise NotImplementedError`.

All this can be fixed but it meant having to modify the experiment configs of other projects and constantly thinking about how things were being resolved which was a bit painful. This new way requires more boilerplate but I like that things are fairly obvious.

Reply:
Now I understand. I think your solution will be cleaner than hacking solutions for each upcoming similar situation 👍